File: kpi/kpi/Ticket.cs
using System;
using System.Collections.Generic;

namespace kpi
{
    public class Ticket
    {
        //поля
        private List<Flight> Way;
        private double TicketPrice;
        private List<Service> IncludedServices;
        private List<Service> AdditionalServices;
        private Tourist Client;

        //конструктор
        public Ticket(List<Flight> flights)
        {
            Way = flights;
            TicketPrice = 0;
            foreach (var flight in Way)
            {
                TicketPrice += flight.FlightPrice;
            }
        }

        //методи
        public void PrintTicket()
        {
            Console.Write("Шлях:");
            for (int i = 0; i < Way.Count-1; i++)
            {
                Console.Write(Way[i].StartPoint+" -> ");
            }
            Console.WriteLine(Way[Way.Count-1].EndPoint);
            Console.WriteLine("Початкова дата: "+Way[0].startDate);
            Console.WriteLine("Кінцева дата: "+Way[0].endDate);
            Console.WriteLine("Цінa: "+TicketPrice+" грн.");
        }

        public void PrintServices()
        {
            Console.WriteLine("Цінa квитка: "+TicketPrice+" грн.");
            Console.WriteLine("Включені послуги:");
            foreach (var service in IncludedServices)
            {
                Console.WriteLine("- "+service.NameOfService);
            }
            
            Console.WriteLine("Додаткові послуги:");
            foreach (var service in AdditionalServices)
            {
                Console.WriteLine("- "+service.NameOfService+" : "+service.ServicePrice+" грн.");
            }
            
        }
        public void ChooseService(int numberOfService)
        {
            IncludedServices.Add(AdditionalServices[numberOfService]);
            TicketPrice += AdditionalServices[numberOfService].ServicePrice;
            AdditionalServices.Remove(AdditionalServices[numberOfService]);
        }
        
        public void AddFlight(){}

        public double GetPrice()
        {
            return TicketPrice;
        }
    }
}

File: Ticket.cpp
#include "Ticket.h"
#include "Flight.h"
#include "Services.h"
        //puts info about just getted seats in flights into file <ID>.txt
        //And creates ticket
        Ticket::Ticket(vector<Flight> flights = vector<Flight>(0), vector<string> reservedSeats = vector<string>(0) )
        {
            for(int i=0; i<reservedSeats.size(); i++){
                ofstream out("DataBase\\" + flights[i].GetFlightID() + ".txt", ios_base::app);
                out << reservedSeats[i] << endl;
            }

            //cout << "Ticket is done!";
            Way = flights;
            SeatsInPlane = reservedSeats;
            TicketPrice = 0;
            for(Flight flight: Way)
            {
                TicketPrice += flight.GetPrice();
            }
        }
        //printing info about ticket onto console and writing into file Ticket.txt
        void Ticket::PrintTicket()
        {
            ofstream out("DataBase\\Tickets.txt", ios_base::app);
            cout << "-----------------------------------------------------\n";
                out << "-----------------------------------------------------\n";
            cout << "Way:\n";
                out << "Way:\n";
            for (int i = 0; i < Way.size(); i++)
            {
                cout << Way[i].GetStartPoint() << " -> ";
                    out << Way[i].GetStartPoint() << " -> ";
            }
            cout << Way[Way.size()-1].GetEndPoint() << endl;
            cout << "Start Time: " << Way[0].GetStartDate() << endl;
            cout << "Arrival Time: " << Way[Way.size()-1].GetEndDate() << endl;
            //cout << "Price: " << TicketPrice << " USD." << endl;
            cout << "-----------------------------------------------------\n";

                out << Way[Way.size()-1].GetEndPoint() << endl;
                out << "Start Time: " << Way[0].GetStartDate() << endl;
                out << "Arrival Time: " << Way[Way.size()-1].GetEndDate() << endl;
                //out << "Price: " << TicketPrice << " USD." << endl;
                out << "-----------------------------------------------------\n";
            out.close();
            PrintServices();
        }
        //used in PrintTicket(). Also writes info into file Ticket.txt
        void Ticket::PrintServices()
        {
            ofstream out("DataBase\\Tickets.txt", ios_base::app);
            cout << "Ticket Price: " << TicketPrice << " USD." << endl;
            cout << "Included Services:" << endl;
                    out << "Ticket Price: " << TicketPrice << " USD." << endl;
                    out << "Included Services:" << endl;
            for(Service service: IncludedServices)
            {
                cout << "- " << service.PrintService() << endl;
                    out << "- " << service.PrintService() << endl;
            }
            cout << "-----------------------------------------------------\n";
                out << "*****************************************************************************************\n";

        }

        void Ticket::ChooseService(int numberOfService)
        {
            IncludedServices.push_back(AdditionalServices[numberOfService]);
            TicketPrice += AdditionalServices[numberOfService].GetPrice();
            AdditionalServices.erase(AdditionalServices.begin() + numberOfService);
        }

        void Ticket::SetServices(vector<Service> ServicesToAdd) {
            IncludedServices = ServicesToAdd;
            for (Service& x : ServicesToAdd) {
                TicketPrice += x.GetPrice();
            }
        }

        void Ticket::AddFlight(){};

        double Ticket::GetPrice()
        {
            return TicketPrice;
        }
